akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ReplicatedEntityProvider.scala

```diff
@@ -28,29 +28,31 @@ object ReplicatedEntityProvider {
    */
   def create[M, E](
       messageClass: Class[M],
+      typeName: String,
       allReplicaIds: JSet[ReplicaId],
       settingsPerReplicaFactory: akka.japi.function.Function3[
         JEntityTypeKey[M],
         ReplicaId,
         JSet[ReplicaId],
         ReplicatedEntity[M, E]]): ReplicatedEntityProvider[M, E] = {
     implicit val classTag: ClassTag[M] = ClassTag(messageClass)
-    apply[M, E](allReplicaIds.asScala.toSet)((key, replica, _) =>
+    apply[M, E](typeName, allReplicaIds.asScala.toSet)((key, replica, _) =>
       settingsPerReplicaFactory(key.asInstanceOf[EntityTypeKeyImpl[M]], replica, allReplicaIds))
   }
 
   /**
    * Scala API:
-   *
+   * @param typeName The type name used in the [[EntityTypeKey]]
    * @tparam M The type of messages the replicated entity accepts
    * @tparam E The type for envelopes used for sending `M`s over sharding
    */
-  def apply[M: ClassTag, E](allReplicaIds: Set[ReplicaId])(
+  def apply[M: ClassTag, E](typeName: String, allReplicaIds: Set[ReplicaId])(
       settingsPerReplicaFactory: (EntityTypeKey[M], ReplicaId, Set[ReplicaId]) => ReplicatedEntity[M, E])
```

Comment:
why is `allReplicaIds` needed as the 3rd parameter of the `settingsPerReplicaFactory` function? Isn't that already known by the user?

Reply:
👍 typically a constant so let's remove it